kernel/Utilities/GlobalIndexing.h

```diff
@@ -133,115 +141,37 @@ class GlobalIndexing {
+    /// \return The global index of the basis function.
+    int getGlobalIndex(const Base::MeshEntity* entity,
                        std::size_t unknown) const {
         logger.assert_debug(unknown < totalNumberOfUnknowns_,
                             "No such unknown %", unknown);
         const Offsets& offset = offsets_[unknown];
-        logger.assert_debug(offset.includedInIndex_,
-                            "Unknown not included % in the index", unknown);
-        const auto basisStart = offset.elementOffsets_.find(element->getID());
-        logger.assert_debug(
-            basisStart != offsets_[unknown].elementOffsets_.end(),
-            "No indices known for element %:%", element->getID(), unknown);
-        return basisStart->second;
-    }
-    int getGlobalIndex(const Base::Face* face, std::size_t unknown) const {
-        logger.assert_debug(unknown < totalNumberOfUnknowns_,
-                            "No such unknown %", unknown);
-        const Offsets& offset = offsets_[unknown];
-        logger.assert_debug(offset.includedInIndex_,
-                            "Unknown not included % in the index", unknown);
-        const auto basisStart = offset.faceOffsets_.find(face->getID());
-        logger.assert_debug(basisStart != offsets_[unknown].faceOffsets_.end(),
-                            "No indices known for face %:%", face->getID(),
-                            unknown);
-        return basisStart->second;
-    }
-    int getGlobalIndex(const Base::Edge* edge, std::size_t unknown) const {
-        logger.assert_debug(unknown < totalNumberOfUnknowns_,
-                            "No such unknown %", unknown);
-        const Offsets& offset = offsets_[unknown];
-        logger.assert_debug(offset.includedInIndex_,
-                            "Unknown not included % in the index", unknown);
-        const auto basisStart = offset.edgeOffsets_.find(edge->getID());
-        logger.assert_debug(basisStart != offsets_[unknown].edgeOffsets_.end(),
-                            "No indices known for edge %:%", edge->getID(),
-                            unknown);
-        return basisStart->second;
-    }
-    int getGlobalIndex(const Base::Node* node, std::size_t unknown) const {
-        logger.assert_debug(unknown < totalNumberOfUnknowns_,
-                            "No such unknown %", unknown);
-        const Offsets& offset = offsets_[unknown];
-        logger.assert_debug(offset.includedInIndex_,
-                            "Unknown not included % in the index", unknown);
-        const auto basisStart = offset.nodeOffsets_.find(node->getID());
-        logger.assert_debug(basisStart != offsets_[unknown].nodeOffsets_.end(),
-                            "No indices known for node %:%", node->getID(),
-                            unknown);
-        return basisStart->second;
+        ConstOffsetVisitor visitor(offset);
+        entity->visitEntity(visitor);
```

Comment:
I am not sure if the name `visitEntity` should be used here. Maybe `beVisited` or so?

Reply:
I have renamed it to `accept`,  which seems to be the standard in all examples for the visitor pattern that I found.